src/arrayMethodReduce.js

```diff
@@ -5,7 +5,21 @@
  */
 function applyCustomReduce() {
   [].__proto__.reduce2 = function(callback, initialValue) {
-    // write code here
+    let accumulator, startIndex;
+
+    if (initialValue === undefined && arguments.length < 2) {
+      accumulator = this[0];
+      startIndex = 1;
+    } else {
+      accumulator = initialValue;
+      startIndex = initialValue === undefined ? 0 : 0;
+    }
```

Comment:
This could be simplified a bit

```js
    let accumulator = initialValue;
    let startIndex = 0;

    if (arguments.length < 2) {
      startIndex = 1;
      accumulator = this[0];
    }
```